github/table_github_code_owner.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+	"strings"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION
+
+func gitHubCodeOwnerColumns() []*plugin.Column {
+	return []*plugin.Column{
+		// Top columns
+		{Name: "repository_full_name", Type: proto.ColumnType_STRING, Description: "The full name of the repository, including the owner and repo name."},
+		// Other columns
+		{Name: "line", Type: proto.ColumnType_INT, Description: "The rule's line number in the CODEOWNERS file.", Transform: transform.FromField("LineNumber")},
+		{Name: "pattern", Type: proto.ColumnType_STRING, Description: "The pattern used to identify what code a team, or an individual is responsible for"},
+		{Name: "users", Type: proto.ColumnType_JSON, Description: "Users responsible for code in the repo"},
+		{Name: "teams", Type: proto.ColumnType_JSON, Description: "Teams responsible for code in the repo"},
+		{Name: "pre_comments", Type: proto.ColumnType_JSON, Description: "Specifies the comments added above a key."},
+		{Name: "line_comment", Type: proto.ColumnType_STRING, Description: "Specifies the comment following the node and before empty lines."},
+	}
+}
+
+//// TABLE DEFINITION
+
+func tableGithubCodeOwner() *plugin.Table {
+	return &plugin.Table{
+		Name:        "github_code_owner",
+		Description: "Individuals or teams that are responsible for code in a repository.",
+		List: &plugin.ListConfig{
+			Hydrate:           tableGitHubCodeOwnerList,
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			KeyColumns:        plugin.SingleColumn("repository_full_name"),
+		},
+		Columns: gitHubCodeOwnerColumns(),
+	}
+}
+
+// // LIST FUNCTION
+type CodeOwnerRule struct {
+	LineNumber  int
+	Pattern     string
+	Users       []string
+	Teams       []string
+	PreComments []string
+	LineComment string
+}
+
+func tableGitHubCodeOwnerList(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("tableGitHubCodeOwnerList")
+	repoFullName := d.KeyColumnQuals["repository_full_name"].GetStringValue()
+	owner, repoName := parseRepoFullName(repoFullName)
+
+	type CodeOwnerRuleResponse struct {
+		RepositoryFullName string
+		LineNumber         int
+		Pattern            string
+		Users              []string
+		Teams              []string
+		PreComments        []string
+		LineComment        string
+	}
+	getCodeOwners := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		var fileContent *github.RepositoryContent
+		var err error
+
+		client := connect(ctx, d)
+		opt := &github.RepositoryContentGetOptions{}
+		// stop on the first found CODEOWNERS file.
+		// NOTE : a repository can have multiple CODEOWNERS files, even if it's invalid
+		// In that case, GitHub uses precedence over these files in the following order : .github/CODEOWNERS, CODEOWNERS, docs/CODEOWNERS
```

Comment:
Can you please also include a link to the documentation where you found this precedence?

Reply:
I don't find any documentation about that :'(.
The precedence, and github behaviour is only based on my tests on a personnal repository.

I looked on github.blog with no success

Reply:
Thanks for checking @aminvielledebatAtBedrock , I couldn't find any docs either, but if they are based off of your test results, that's good enough for me.